packages/flutter/lib/src/material/switch.dart

```diff
@@ -88,8 +88,18 @@ class Switch extends StatefulWidget {
         assert(inactiveThumbImage != null || onInactiveThumbImageError == null),
         super(key: key);
 
-  /// Creates a [CupertinoSwitch] if the target platform is iOS, creates a
-  /// material design switch otherwise.
+  /// Creates an adaptive [Switch] based on the target platform, following
+  /// Material design's
+  /// [Cross-platform guidelines](https://material.io/design/platform-guidance/cross-platform-adaptation.html).
+  ///
+  /// {@template flutter.material.switch.adaptive}
+  /// The Material design guidelines recommend using the native switches on iOS
```

Comment:
Although the Material design guidelines do in fact recommend using native iOS switches, we clearly do not do that. The most that we can claim here, is that we'll create switches that resemble the native iOS versions.

Reply:
updated.